File: frontend/src/types/dog.ts
/**
 * 백엔드 공통 응답 구조
 */
export interface ApiResponse<T> {
  success: boolean;
  message: string | null;
  data: T;
}

/**
 * 반려견 정보 (응답용)
 */
export interface Dog {
  id: number;
  userId: number;
  name: string;
  breed: string | null;
  birthDate: string | null;
  weight: number | null;
  profileImageUrl: string | null;
  createdAt: string;
  updatedAt: string;
}

/**
 * 반려견 생성 요청 DTO
 */
export interface DogCreateRequest {
  name: string;
  breed: string | null;
  birthDate: string | null; // YYYY-MM-DD
  weight: number | null;
  profileImageUrl: string | null;
}

/**
 * 반려견 수정 요청 DTO
 */
export interface DogUpdateRequest extends DogCreateRequest {}

File: backend/src/main/java/com/pawcarechart/backend/dog/entity/Dog.java
package com.pawcarechart.backend.dog.entity;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Entity
@Table(name = "dogs")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Dog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(nullable = false, length = 50)
    private String name;

    @Column(length = 50)
    private String breed;

    @Column(name = "birth_date")
    private LocalDate birthDate;

    private Double weight;

    @Column(name = "profile_image_url", columnDefinition = "TEXT")
    private String profileImageUrl;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @Builder
    public Dog(Long userId, String name, String breed, LocalDate birthDate, Double weight, String profileImageUrl) {
        this.userId = userId;
        this.name = name;
        this.breed = breed;
        this.birthDate = birthDate;
        this.weight = weight;
        this.profileImageUrl = profileImageUrl;
    }

    @PrePersist
    protected void onCreate() {
        LocalDateTime now = LocalDateTime.now();
        this.createdAt = now;
        this.updatedAt = now;
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }

    public void update(String name, String breed, LocalDate birthDate, Double weight, String profileImageUrl) {
        this.name = name;
        this.breed = breed;
        this.birthDate = birthDate;
        this.weight = weight;
        this.profileImageUrl = profileImageUrl;
    }
}
